packages/select/src/vaadin-select.js

```diff
@@ -641,6 +723,43 @@ class Select extends DelegateFocusMixin(FieldMixin(SlotMixin(ElementMixin(Themab
     return !(this.invalid = !(this.disabled || !this.required || this.value));
   }
 
+  /**
+   * Computes the final renderer for the `__renderer` property.
+   * @return {!SelectRenderer}
+   * @private
+   */
+  __computeRenderer(renderer) {
+    if (renderer) {
+      return renderer;
+    }
+
+    return this.__defaultRenderer;
+  }
+
+  /**
+   * Renders items provided by the `items` property.
+   * @param {HTMLElement} root
+   * @param {Select} _select
+   * @private
+   */
+  __defaultRenderer(root, _select) {
```

Comment:
This could be extracted into a function, `this.items` can be replaced with `select.items`. IMO we are putting too much stuff on a single prototype in general. Here is an easy opportunity to improve on that.

Reply:
If we prefer listing `__defaultRenderer` as one of the dependencies of the `__computedRenderer` observer, see https://github.com/vaadin/web-components/pull/2990#discussion_r745429047, it actually becomes quite challenging to extract it to outside. :)

Reply:
... and did you mean the readability or some possible performance issues?

Reply:
Decided to leave the method inside the class, there is not that much sense to extract it since it utilizes some of the internal API.